src/recipes.jl

```diff
@@ -213,21 +221,22 @@ function Makie.plot!(gp::GraphPlot)
                           linewidth=gp.edge_width,
                           gp.edge_attr...)
 
-    # plott arrow heads
-    arrow_pos = @lift if !isempty($edge_paths)
-        broadcast(interpolate, $edge_paths, $(gp.arrow_shift))
+    # plot arrow heads
+    arrow_shift = @lift update_arrow_shift!(graph[], gp, $edge_paths, $(gp.arrow_shift), $to_angle, $to_point)
```

Comment:
I think the function is not mutating so shouldn't have a `!`. A mutating version with observables would look something like that:
```julia
arrow_shift = Observable(zeros(Float32, ne(gp))
onany(edge_paths, gp.arrow_shift, topoint) do ep, as, tp
    update_arrow_shift!(arrow_shift, graph[], ep, as, toangle[], tp)
end
```
(in that case you'd need to resize the `arrow_shift` to `ne(graph)` within the function)
Also you don't need to add an explicit observable-dependency on `toangle`, because if `toangle` updates it will trigger `topoint` and otherwise you'll trigger this function twice.

Reply:
thanks. fixed

Reply:
I tried removing the `$` from `to_angle`, but get the error:
ERROR: MethodError: objects of type Observable{GraphMakie.var"#1906#1926"{GraphMakie.var"#1904#1924"{Scene}}} are not callable

This happens because I call the function to_angle inside the `update_arrow_shift` function. How do we deal with this so that we don't trigger the function twice?

Reply:
Ahh, never mind. I see the bracket next to it so that the function is passed and not the observable. Fixed it now.